spec/index.bs

```diff
@@ -114,33 +168,131 @@ interface GPUAdapter : GPUObjectBase {
 };
 </script>
 
+{{GPUAdapter}} has the following attributes:
+
+<dl dfn-type=attribute dfn-for=GPUAdapter>
+    : <dfn>\[[adapter]]</dfn>, of type [=adapter=], readonly
+    ::
+        An internal slot holding the [=adapter=] to which this {{GPUAdapter}}
+        refers.
+
+    : <dfn>name</dfn>
+    ::
+        A human-readable name identifying the adapter.
+        The contents are implementation-defined.
+
+    : <dfn>extensions</dfn>
+    ::
+        A {{GPUExtensions}} object which enumerates the extensions supported
+        by the user agent, and whether each extension is supported by the
+        underlying implementation.
+          - If an extension is not supported by the user agent,
+            it will not be present in the object.
+          - If an extension is supported by the user agent, but
+            not by the [=adapter=], it will be `false`.
+          - If an extension is supported by the user agent and
+            by the [=adapter=], it will be `true`.
+</dl>
+
 ### Getting an Adapter ### {#adapter-creation}
 
-<script type=idl>
-[Exposed=Window]
-interface GPU {
-    // May reject with DOMException  // TODO: DOMException("OperationError")?
-    Promise<GPUAdapter> requestAdapter(optional GPURequestAdapterOptions options);
-};
-</script>
+To get a {{GPUAdapter}}, use {{GPU/requestAdapter()}}.
 
 <script type=idl>
 dictionary GPURequestAdapterOptions {
-    GPUPowerPreference? powerPreference;
+    GPUPowerPreference powerPreference;
 };
 </script>
 
+<dfn dictionary>GPURequestAdapterOptions</dfn>
+provides requirements and hints to the user agent indicating what
```

Comment:
is "user agent" a proper term for "browser WebGPU implementation"?

Reply:
user agent (== browser in this case) encompasses its WebGPU implementation, so I think so.

Reply:
Uh, I find this quite confusing. The semantics of being an "agent" of the user is quite far from all the machinery that *implements* the API, which is what we mean in most cases by "user agent"...